src/parser.rs

```diff
@@ -2562,10 +2568,38 @@ impl<'a> Parser<'a> {
             None
         };
         match self.next_token() {
+            Token::Word(w)
+                if (w.keyword == Keyword::KEY || w.keyword == Keyword::INDEX)
+                    && dialect_of!(self is MySqlDialect) =>
+            {
+                let name = Some(self.parse_identifier()?);
+                let columns = self.parse_parenthesized_column_list(Mandatory)?;
+                match w.keyword {
+                    Keyword::KEY => Ok(Some(TableConstraint::Key {
+                        name,
+                        columns,
+                        format: KeyFormat::Key,
+                    })),
+                    Keyword::INDEX => Ok(Some(TableConstraint::Key {
+                        name,
+                        columns,
+                        format: KeyFormat::Index,
+                    })),
+                    _ => unreachable!(),
+                }
+            }
             Token::Word(w) if w.keyword == Keyword::PRIMARY || w.keyword == Keyword::UNIQUE => {
                 let is_primary = w.keyword == Keyword::PRIMARY;
                 if is_primary {
                     self.expect_keyword(Keyword::KEY)?;
+                } else if dialect_of!(self is MySqlDialect) {
```

Comment:
Please add the `GenericDialect` here. The idea of that, as long as I understood by [this](https://github.com/sqlparser-rs/sqlparser-rs/pull/642#discussion_r985143947) comment, is to support everything possible.

Reply:
It's hard to cover all dialects now. so using the specific dialect to avoid affecting other dialects.

Reply:
@RainJoe THe`GenericDialect` is the one that contains literally EVERY single possible AST structure. Therefore, it should contemplate any possible query

`dialect_of!(self is GenericDialect | MySqlDialect)` should be enough